### ISSUE TITLE
patches created with v5 not compatible with v6 ?

### ISSUE THREAD

**Issue by vesper8** [ds300/patch-package#118]
A few weeks ago I used this package for the first time

I cloned https://github.com/vuex-orm/plugin-axios
applied the changes from this PR: vuex-orm/plugin-axios#8

and then created a patch

Today I wanted to use this patch on a new project so I installed patch-package and postinstall-postinstall on this new project, added the patch-package command to my postinstall script, and copied over the patch I created a few weeks ago

I kept getting this error when running 'yarn':

```
Error: Command failed: yarn run postinstall
/DEV/my-project/node_modules/patch-package/dist/patch/parse.js:281
        throw new Error("Unexpected file mode string: " + mode);
        ^

Error: Unexpected file mode string: 120000
    at parseFileMode (/DEV/my-project/node_modules/patch-package/dist/patch/parse.js:281:15)
    at interpretParsedPatchFile (/DEV/my-project/node_modules/patch-package/dist/patch/parse.js:243:27)
    at Object.parsePatchFile (/DEV/my-project/node_modules/patch-package/dist/patch/parse.js:296:20)
    at Object.exports.applyPatch (/DEV/my-project/node_modules/patch-package/dist/applyPatches.js:100:25)
    at /DEV/my-project/node_modules/patch-package/dist/applyPatches.js:57:21
    at Array.forEach (<anonymous>)
    at Object.exports.applyPatchesForApp (/DEV/my-project/node_modules/patch-package/dist/applyPatches.js:42:11)
    at Object.<anonymous> (/DEV/my-project/node_modules/patch-package/dist/index.js:46:24)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
error Command failed with exit code 1.

    at checkExecSyncError (child_process.js:616:11)
    at execSync (child_process.js:653:13)
    at Object.<anonymous> (/DEV/my-project/node_modules/postinstall-postinstall/run.js:14:5)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
```

However applying the patch manually with git apply did run without errors

I then realized that a few weeks ago when I installed patch-package it installed 
```
"patch-package": "^5.1.1",
```

and today it installed 

```
"patch-package": "^6.0.2",
```

After reverting to ^5.1.1 I am now able to use my patch

So.. patches created with v5 don't work on v6? Is this normal? I couldn't find any changelog or notes regarding v6.. would be nice to know if I can remain on v5 or if v6 has some useful new things that I would want. Is there a way to "upgrade" patches or would I have to re-apply the patch manually using v6?

Thanks!

**Comment by ds300** [comment]
Hey! The changelog is at `/CHANGELOG.md` but it doesn't mention breaking changes because I was hoping that there weren't any.

Weird that you have a file mode annotation of `120000`.

According to this: https://medium.com/@tahteche/how-git-treats-changes-in-file-permissions-f71874ca239d

> 100755 for executable and 100644 for non-executable are the only file modes Git records.

Maybe the parser was parsing the file wrong, can you attach a copy of the patch file here? Also what gets printed when you run `ls -l ./path/to/the/relevant/file`?

**Comment by vesper8** [comment]
this is the result of ls -l

```
ls -l patches/@vuex-orm
total 152584
-rw-r--r--@ 1 vesperknight  staff  78122251 14 Jan 00:46 plugin-axios+0.2.2.patch
```

using OSX Mojave, git version 2.20.1

here is a copy of the patch:

https://www.dropbox.com/sh/127ga050hmaaptr/AADNTCAuaAmpsCPsK8DOzoeOa?dl=0

As I mentioned in my other issue, it's big. It's 78mb

I answered the other issue about a detailed description of how I created this large patch

Here are the details of my package.json in relation to this patch:

```
    "@vuex-orm/core": "^0.31.2",
    "@vuex-orm/plugin-axios": "^0.2.2",
```

As I mentioned above, if I use this in my package.json

```
    "patch-package": "^5.1.1",
    "postinstall-postinstall": "^2.0.0",
```

and then run 'yarn' which includes the postinstall patch-package directive, then it works perfectly

If I use

```
"patch-package": "^6.0.2",
```

however, then I get the error. I hope you can reproduce it with the attached patch

**Comment by vesper8** [comment]
this is the result of ls -l

```
ls -l patches/@vuex-orm
total 152584
-rw-r--r--@ 1 vesper8  staff  78122251 14 Jan 00:46 plugin-axios+0.2.2.patch
```

using OSX Mojave, git version 2.20.1

here is a copy of the patch:

https://www.dropbox.com/sh/127ga050hmaaptr/AADNTCAuaAmpsCPsK8DOzoeOa?dl=0

As I mentioned in my other issue, it's big. It's 78mb

I answered the other issue about a detailed description of how I created this large patch

Here are the details of my package.json in relation to this patch:

```
    "@vuex-orm/core": "^0.31.2",
    "@vuex-orm/plugin-axios": "^0.2.2",
```

As I mentioned above, if I use this in my package.json

```
    "patch-package": "^5.1.1",
    "postinstall-postinstall": "^2.0.0",
```

and then run 'yarn' which includes the postinstall patch-package directive, then it works perfectly

If I use

```
"patch-package": "^6.0.2",
```

however, then I get the error. I hope you can reproduce it with the attached patch

**Comment by vesper8** [comment]
so.. as a result of the way I applied the patch it created this symlink problem?

Or is this a result of the fact that I access my dev folder and all my dev projects via symlinks ?

And so this symlink problem was only introduced as of v6 ? How come it works on v5?

**Comment by ds300** [comment]
> so.. as a result of the way I applied the patch it created this symlink problem?
> Or is this a result of the fact that I access my dev folder and all my dev projects via symlinks ?

I think the problem was the way you created the patch file, but not sure. If you try again with the method I suggested in #117 and it still fails in v6 then maybe it's because of your dev setup.

> And so this symlink problem was only introduced as of v6 ? How come it works on v5?

v5 uses git to apply patches, v6 uses a custom TypeScript implementation which doesn't support creating symlinks.

**Comment by ds300** [comment]
Just released specific logic to handle this case in 6.1.2. Thanks again for the report ❤️
